trie/hexary.py

```diff
@@ -243,7 +268,34 @@ def _prune_node(self, node):
 
         # Prune only if no exception is raised
         if should_prune:
-            del self.db[prune_key]
+            self._prune_key(prune_key)
+
+    def _prune_key(self, key):
+        self.ref_count[key] -= 1
```

Comment:
Maybe this should have an existence check since if the key were to not be in the database this would end up with the refcount being negative.